website/content/en/docs/about/under-the-hood/architecture/buffering-model.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Buffering model
+weight: 2
+tags: ["buffering", "buffers", "disk"]
+---
+
+Vector implements a buffering model that allows operators to choose whether to prioritize
+performance or durability when handling an excess of events beyond what a sink can process.
+
+## Backpressure and the need to buffer
+
+While operators typically strive to ensure their Vector deployments are appropriately sized for the
+expected load, sometimes problems can occur. Maybe an application starts generating more logs than
+normal, or the downstream service where you're sending data starts to respond slower than normal.
+
+Part of Vector's topology design involves propagating _backpressure_, which is a signal that events cannot be
+processed as quickly as they are being received. When one component tries to send more events to a
+component than that component can currently handle, the sending component is informed of this
+indirectly. Backpressure can travel all the way from a sink, up through transforms, back to a
+source, and ultimately, even to clients such as applications sending logs over HTTP.
```

Comment:
```suggestion
indirectly. Backpressure can travel all the way from a `sink`, up through `transforms`, back to a
`source`, and ultimately, even to clients such as applications sending logs over HTTP.
```

We haven't written a writing style guide (AFAIK), but does it make sense to format these "vector internals" specially?

Reply:
Hmmm, perhaps.

I'm not sure using the monospace code styling is the best thing. I'd almost want/expect them to be hyperlinks back to the glossary, perhaps?

Reply:
Yeah, no strong feelings on how - I think we might monospace them elsewhere today.

Reply:
I'm going to crib from the concurrency model page where we hyperlink "source" to the top-level configuration section for sources. Just a hyperlink, though: it doesn't monospace them.